src/faultlog/unresolved_pels.cpp

```diff
@@ -0,0 +1,253 @@
+#include <attributes_info.H>
+#include <fmt/format.h>
+
+#include <libguard/guard_interface.hpp>
+#include <phosphor-logging/log.hpp>
+#include <unresolved_pels.hpp>
+#include <util.hpp>
+
+extern "C"
+{
+#include <libpdbg.h>
+}
+
+namespace openpower::faultlog
+{
+using ::nlohmann::json;
+using ::phosphor::logging::level;
+using ::phosphor::logging::log;
+
+using PropertyValue =
+    std::variant<std::string, bool, uint8_t, int16_t, uint16_t, int32_t,
+                 uint32_t, int64_t, uint64_t, double>;
+
+using Properties = std::map<std::string, PropertyValue>;
+
+using Interfaces = std::map<std::string, Properties>;
+
+using Objects = std::map<sdbusplus::message::object_path, Interfaces>;
+
+constexpr auto STATE_CONFIGURED = "CONFIGURED";
+constexpr auto STATE_DECONFIGURED = "DECONFIGURED";
+
+struct GuardedTarget
+{
+    pdbg_target* target = nullptr;
+    std::string phyDevPath = {};
+    GuardedTarget(const std::string& path) : phyDevPath(path)
+    {}
+};
+
+static int getGuardedTarget(struct pdbg_target* target, void* priv)
+{
+    // recursive callback function that exits when the target matching the
+    // guarded targets physical path is found in the device tree.
+    // to exit the recusive function return 1 to continue return 0
+    GuardedTarget* guardTarget = reinterpret_cast<GuardedTarget*>(priv);
+    ATTR_PHYS_DEV_PATH_Type phyPath;
+    if (!DT_GET_PROP(ATTR_PHYS_DEV_PATH, target, phyPath))
+    {
+        if (strcmp(phyPath, guardTarget->phyDevPath.c_str()) == 0)
+        {
+            guardTarget->target = target;
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void UnresolvedPELs::populate(sdbusplus::bus::bus& bus,
+                              GuardRecords& guardRecords,
+                              std::vector<uint32_t>& processedEid,
+                              json& jsonNag)
+{
+    log<level::INFO>("UnresolvedPELs::populate");
```

Comment:
Since this is a new application, I'd recommend switching over to lg2.  It's a lot less verbose as you can just do lg2::info() or info().  It's also easier to pass in params.  Just a suggestion though.

Reply:
yeah good suggestion, will switch to it, as it will specify the function and place it failed.